airbyte-scheduler/app/src/test/java/io/airbyte/scheduler/app/JobSubmitterTest.java

```diff
@@ -226,4 +230,68 @@ void testMDC() throws Exception {
     assertTrue(MDC.getCopyOfContextMap().isEmpty());
   }
 
+  @Nested
+  class OnlyOneJobIdRunning {
+
+    @Test
+    public void testMultipleSubmissionShouldFail() throws Exception {
```

Comment:
i think the name of this one is a bit misleading or maybe add a comment explaining what you're testing. this was a bit hard for me to parse. if i'm understanding it correctly it is testing that while a job is running or queued, the submitter still runs but if it tries to submit the same job again subsequent submissions don't happen. thus `getNextJob` happens more than once but `submitJob` should happen exactly once.

Reply:
Exactly that. I will add comments.